app/src/main/java/com/amos/flyinn/settingsCheck/settingsCheck.java

```diff
@@ -0,0 +1,116 @@
+package com.amos.flyinn.settingsCheck;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.os.StrictMode;
+import android.provider.Settings;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.amos.flyinn.MainActivity;
+import com.amos.flyinn.summoner.ADBService;
+import com.tananaev.adblib.AdbBase64;
+import com.tananaev.adblib.AdbConnection;
+import com.tananaev.adblib.AdbCrypto;
+
+import java.io.Console;
+import java.io.IOException;
+import java.net.Socket;
+import java.util.HashSet;
+import java.util.List;
+
+import static android.support.v4.content.ContextCompat.startActivity;
+
+public class settingsCheck {
+
+    public enum SettingsType {
+        Developer, ADBperUSB, ADBperNetwork
+    }
+
+    private Context context;
+    private ADBService service = new ADBService();
+    public settingsCheck(Context context) {
+        this.context=context;
+    }
+
+    /**
+     * Collect all disabled settings which are necessary to be enabled
+     * for this app in a list
+     * @return List<Settings> List of disabled settings.
+     */
+    public HashSet<SettingsType> GetMissingSettings() {
```

Comment:
Please create a unittest demonstating this in robelectric. If possible please also mock cases in which settings are changed or not existing at all. Please check whether this is reasonable, if it is a lot of work, we can leave out these cases for now and add them later in integration testing.